src/extension/features/budget/cover-overspending-from-future/index.js

```diff
@@ -0,0 +1,163 @@
+import { Feature } from 'toolkit/extension/features/feature';
+import { /* getSelectedMonth, */ getEntityManager, isCurrentRouteBudgetPage } from 'toolkit/extension/utils/ynab';
+import { formatCurrency } from 'toolkit/extension/utils/currency';
+import { l10n } from 'toolkit/extension/utils/toolkit';
+import { controllerLookup } from 'toolkit/extension/utils/ember';
+
+const BUTTON_CLASS = 'tk-button-cover-from-future';
+
+export class CoverOverspendingFromFuture extends Feature {
+  injectCSS() {
+    return require('./index.css');
+  }
+
+  shouldInvoke() {
+    return isCurrentRouteBudgetPage();
+  }
+
+  invoke() {
+    this.verifyButtonDisplay();
+  }
+
+  observe(changedNodes) {
+    if (!this.shouldInvoke()) return;
+    if (changedNodes.has('budget-table-row is-sub-category is-checked')) {
+      this.invoke();
+    }
+  }
+
+  verifyButtonDisplay() {
+    const checkedRows = controllerLookup('budget').get('checkedRows');
+
+    // Do nothing with the button if more than one row is selected. (maybe change in the future)
+    if (checkedRows.length !== 1) {
+      return this.removeButton();
+    }
+
+    const category = checkedRows[0];
+    const { isMasterCategory, available, budgeted } = category.getProperties('isMasterCategory', 'available', 'budgeted');
+
+    // TODO: Test
+    console.log(budgeted);
+
+    // We can't budget for master categories
+    // We don't need this if available is not negative
+    if (isMasterCategory || available >= 0) {
+      return this.removeButton();
+    }
+
+    // Search for the button
+    let button = $(`.${BUTTON_CLASS}`);
+
+    // Create button if it doesn't exist already
+    if (!button.length) {
+      button = $('<button>', {
+        class: `budget-inspector-button ${BUTTON_CLASS}`
+      }).click(this.coverFromFuture.bind(this, category));
+
+      // Add the button at the top.
+      $('.inspector-quick-budget button').first().before(button);
```

Comment:
as a personal preference, I would rather see this get added to the bottom :P I'm already really used to moving one category at a time and pressing the first button. haha.

Also, looking at the existing grouping of `quick-budget` options, I believe this makes most sense in the second group.

What I would do is:

```
$('.budget-inspector .option-groups').children().last().append(button)
```
targeting the inspector, find the `option-groups` div, select the last child from that query and append our button :D

Reply:
unfortunately there's no class on the children of `option-groups`